tests/test_main.py

```diff
@@ -339,3 +339,56 @@ def test_main_noxfile_options_sessions():
         # Verify that the config looks correct.
         config = honor_list_request.call_args[1]["global_config"]
         assert config.sessions == ["test"]
+
+
+@pytest.mark.parametrize(("isatty_value", "expected"), [(True, True), (False, False)])
+def test_main_color_from_istty(isatty_value, expected):
+    sys.argv = [sys.executable]
+    with mock.patch("nox.workflow.execute") as execute:
+        execute.return_value = 0
+        with mock.patch("sys.stderr.isatty") as istty:
+            istty.return_value = isatty_value
+
+            # Call the main function.
+            with mock.patch.object(sys, "exit"):
+                nox.__main__.main()
+
+            config = execute.call_args[1]["global_config"]
```

Comment:
Whenever I see something like `execute.call_args[1]["global_config"]` my brain is like "there is so much state **not** being verified". Makes me uneasy. Thoughts?

Reply:
I think the rest of the state is being verified well elsewhere. `main` basically just invokes the argument parser and hands the rest off to `workflow.execute`. We could refactor this at another time to make main a bit easier to test, but this is fine for now.